input/docs/declarative.md

```diff
@@ -0,0 +1,448 @@
+different clusters.  The `skupper` command uses your kubeconfig and
+current context to select the namespace where it operates.
+
+To avoid getting your wires crossed, you must use a distinct
+kubeconfig or context for each namespace.  The easiest way is to use
+separate console sessions.
+
+### Configure separate console sessions
+
+Start a console session for each of your namespaces.  Set the
+`KUBECONFIG` environment variable to a different path in each session.
+
+<div class="code-label session-2">Console for West</div>
+
+    export KUBECONFIG=$HOME/.kube/config-west
+
+<div class="code-label session-1">Console for East</div>
+
+    export KUBECONFIG=$HOME/.kube/config-east
+
+### Log in to your clusters
+
+The methods for logging in vary by Kubernetes provider.  Find the
+instructions for your chosen provider or providers and use them to
+authenticate and establish access for each console session.
+
+<div class="code-label session-2">Console for West</div>
+
+    $ <login-command-for-your-provider>
+
+<div class="code-label session-1">Console for East</div>
+
+    $ <login-command-for-your-provider>
+
+See the following links for more information:
+
+<ul class="column-list">
+  <li><a href="minikube.html#logging-in">Minikube</a></li>
+  <li><a href="https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html">Amazon Elastic Kubernetes Service</a></li>
+  <li><a href="https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough#connect-to-the-cluster">Azure Kubernetes Service</a></li>
+  <li><a href="https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl">Google Kubernetes Engine</a></li>
+  <li><a href="https://docs.openshift.com/container-platform/4.1/cli_reference/getting-started-cli.html#cli-logging-in_cli-developer-commands">Red Hat OpenShift</a> or <a href="https://docs.okd.io/latest/cli_reference/get_started_cli.html#basic-setup-and-login">OKD</a></li>
+</ul>
+
+### Set the current namespaces
+
+Use `kubectl create namespace` to create the namespaces you wish to
+use.  Use `kubectl config set-context` to set the current namespace
+for each session.
+
+<div class="code-label session-2">Console for West</div>
+
+    kubectl create namespace west
+    kubectl config set-context --current --namespace west
+
+<div class="code-label session-1">Console for East</div>
+
+    kubectl create namespace east
+    kubectl config set-context --current --namespace east
+
+
+## Step 2: Create a Skupper site in each namespace
+
+
+### Deploy the site controller
+
+<div class="code-label session-2">West</div>
+
+    $ oc apply -f https://raw.githubusercontent.com/skupperproject/skupper/0.3/cmd/site-controller/deploy-watch-current-ns.yaml
+
+<div class="code-label session-1">East</div>
+
+    $ oc apply -f https://raw.githubusercontent.com/skupperproject/skupper/0.3/cmd/site-controller/deploy-watch-current-ns.yaml
+
+**NOTE:** To configure Skupper for all namespaces in a cluster:
```

Comment:
Would it be useful to highlight that to watch all will require cluster wide permissions for the user?

Reply:
agreed